src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReportingService.kt

```diff
@@ -0,0 +1,34 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.authorization.ServiceProviderAccessScopeMapper
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.ReferralReportDataDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
+import java.time.OffsetDateTime
+import java.util.*
+import javax.transaction.Transactional
+
+@Service
+@Transactional
+class ReportingService(
+  val referralRepository: ReferralRepository,
+  val serviceProviderUserAccessScopeMapper: ServiceProviderAccessScopeMapper,
+  @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
+  @Value("\${interventions-ui.locations.referral-details}") private val interventionsUIReferralDetailsLocation: String,
+) {
+  fun getReportData(to: OffsetDateTime, from: OffsetDateTime, user: AuthUser): List<ReferralReportDataDTO> {
```

Comment:
nit: `from, to` is a more natural order for these params

nit: it's not clear what 'user' is for in the param name. this makes me wonders, is it the responsibility of the reporting service to look up the access scope? could this method signature be like "getReportDataForContracts" and leave that up to the caller? that way if, down the line, we want to adds methods to say, just get report data for cases assigned to a certain user, we could add a method like 'getReportDataForReferralAssignee' that takes a user param. 

we don't have to do it now, because it doesn't affect the API, but it's good to consider how this class might evolve, and how the method signatures look to uninformed future readers.